schemaregistry/schemaregistry.md

```diff
@@ -0,0 +1,257 @@
+  related set of schemas, typically belonging to a particular application or
+  subject to shared access control rules.
+- Schema: A document describing the structure, names, and types of some
+  structured data payload.
+- Version: A specific version of a schema document. Even though not prescribed
+  in this specification, an implementation might choose to impose compatibility
+  constraints on versions following the initial version of a schema.
+
+> NOTE: There are two fundamentally different options for how we can organize
+> multiple schema formats and this specification draft reflects both: 
+>
+> 1) In an "ideal" REST service, any data entity can be represented ("R" in
+>    REST) in multiple different data formats, which a client can request in
+>    prioritized order during content-type negotiation. If a service offers up
+>    the same entity in JSON, Protobuf, and Avro formats, there would therefore
+>    be three distinct schemas, each defining a representation of the data
+>    entity for the respective format. From the same "REST" mindset perspective,
+>    the ideal representation of this set of schemas, which differ with regards
+>    to the format they describe, but ought to be semantically identical, is to
+>    also differentiate them only by their content type and maintain them under
+>    the exact same name as if they were one schema.
+> 2) Since the above strategy is truly RESTful, but quite esoteric if you've not
+>    grown up as a RESTafarian, the alternative strategy for concurrently
+>    handling multiple schema formats is much simpler: Constrain each schema
+>    group to a single format.
+
+## 2. Schema Registry Elements
+
+This section further describes the elements enumerated in the introduction.
+
+### 2.1. Schema Group
+
+A schema group is a named collection of schemas. Since a schema registry is
+often a resource with a scope greater than a single application and might even
+span multiple organizations, it is very useful to put a grouping construct
+around sets of schemas that are related either by ownership or by a shared
+subject matter context.
+
+Implementations of this specification MAY associate access control rules with
+schema groups. For instance, a user or a group of users might be given write
+access only to a particular schema group that their organization owns. If trade
+secret protection is required for an application or parts of it and the schema
+structure would give some of those away, read access to a group of schemas might
+likewise be restricted. 
+
+Access control rules at the schema group level MUST NOT limit visibility of the
+schema groups themselves to users authorized to enumerate schema groups at the
+registry level or to create new schema groups in order to create unambiguous
+transparency about which `id` have already been taken. An implementation MAY
+withhold all details other than the `id` attribute from a user not authorized
+to access the schema group.
```

Comment:
Section 3.1 says that this `id` will be used as part of the URL to access a particular schema. Does the MUST NOT in this paragraph imply some kind of security leak by requiring everyone who can at least see schema groups in general to also be able to see the IDs of schema groups they don't have access to?

Reply:
Unauthorized parties can find out by probing the registry for conflicts, so there is no significant security gain in hiding the group names. This mandates that the API doesn't try to obscure the reason for potential conflicts.